backend/games/word_guesser/WordGuesser.py

```diff
@@ -0,0 +1,112 @@
+from random import randint
+from typing import Dict, List
+
+MAX_GUESSES = 6
+
+# initialize valid guesses and possible solutions
+SOLUTIONS: str
+VALID_WORDS: str
+
+solutions_file = open("backend/games/word_guesser/WordGuesser_Solutions.txt")
+guesses_file = open("backend/games/word_guesser/WordGuesser_ValidGuesses.txt")
+SOLUTIONS = solutions_file.read()
+SOLUTION_LIST = SOLUTIONS.splitlines()
+
+VALID_WORDS = guesses_file.read()
+VALID_WORDS_LIST = VALID_WORDS.splitlines()
+
+
+class WordGuesser:
+    def __init__(self):
+        self.guesses = {"guess": "", "results": []}
+        self.game_board: List[str] = []
+        self.progress: Dict = {}
+        self.goal_word = "basic"
```

Comment:
I was having trouble figuring out how your tests pass, then I realized that in your tests you are not calling `game.start_game()`, so then `"basic"` never gets replaced as the goal word.

That is clever, but it would be easier if we can specify any goal word so that we can have deterministic tests. You can add an optional argument to the constructor and use that word as the goal word if present. Then, when you call `game.start_game()`, if the word is not set, you can choose a random one, or if it is set, keep it.

Reply:
This makes more sense